lib/rspec/expectations/configuration.rb

```diff
@@ -141,14 +149,43 @@ def self.format_backtrace(backtrace)
       # Configures whether RSpec will warn about matcher use which will
       # potentially cause false positives in tests.
       #
-      # @param value [Boolean]
-      attr_writer :warn_about_potential_false_positives
+      # @param [Boolean] boolean
+      def warn_about_potential_false_positives=(boolean)
+        if boolean
+          self.on_potential_false_positives = :warn
+        elsif warn_about_potential_false_positives?
```

Comment:
As the value could be `:raise` and `warn_about_potential_false_positives = false` shouldn't change that, so we only change it to `:nothing` when it's `:warn` already.